### PR TITLE
feat(multi-tenant): use cached backend config

### PR DESCRIPTION
**Pull request by kuldeep0020**
# Use cached config backend to reduce the load on config backend. [Reference](https://github.com/rudderlabs/rudder-config-backend/blob/18169c409dc9b0dce137c7467da99b26112af758/src/swagger3/paths/dataPlanePaths.yaml#L166)

## Notion Ticket

[Notion Link](https://www.notion.so/rudderstacks/Use-cached-endpoint-for-hosted-configs-11a9ce59c6fa4917a94ad2a363c733ba)

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: config/backend-config/multitenant_workspaces.go [hunk 1]
```diff
@@ -82,7 +82,11 @@ func (workspaceConfig *MultiTenantWorkspacesConfig) getFromAPI(workspaceArr stri
 	var url string
 	// TODO: hacky way to get the backend config for multi tenant through older hosted backed config
 	if config.GetBool("BackendConfig.useHostedBackendConfig", false) {
-		url = fmt.Sprintf("%s/hostedWorkspaceConfig?fetchAll=true", configBackendURL)
+		if config.GetBool("BackendConfig.cachedHostedWorkspaceConfig", true) {
```
Comment (cisse21) [inline config/backend-config/multitenant_workspaces.go:85]: Let's also have this check in hosted_workspaces.go

**Review by cisse21 — CHANGES_REQUESTED** [review]

File: config/backend-config/hosted_workspaces.go [hunk 2]
```diff
@@ -79,7 +79,12 @@ func (multiWorkspaceConfig *HostedWorkspacesConfig) GetWorkspaceLibrariesForWork
 
 //Get returns sources from all hosted workspaces
 func (multiWorkspaceConfig *HostedWorkspacesConfig) Get(_ string) (ConfigT, bool) {
-	url := fmt.Sprintf("%s/hostedWorkspaceConfig?fetchAll=true", configBackendURL)
+	var url string
+	if config.GetBool("BackendConfig.cachedHostedWorkspaceConfig", true) {
+		url = fmt.Sprintf("%s/cachedHostedWorkspaceConfig", configBackendURL)
```
Comment (atzoum) [inline config/backend-config/hosted_workspaces.go:84]: Why do we need a separate URL for that? We should use standard HTTP caching constructs, e.g. use the same URL and control caching through the `cache-control` request header.

**Review by cisse21 — APPROVED** [review]

File: config/backend-config/hosted_workspaces.go (hunk 2 above)
Comment (cisse21) [reply config/backend-config/hosted_workspaces.go:84]: Not sure why but a new endpoint has been implemented by the control-plane team, Ref : [here](https://rudderlabs.slack.com/archives/C02AASYS6R1/p1653293448145139?thread_ts=1653292691.828539&cid=C02AASYS6R1)

Comment (lvrach) [reply config/backend-config/hosted_workspaces.go:84]: That would be ideal, but the control-plane team has already implemented the endpoint like this. We have started a discussion to improve those endpoints overall, but for now, is important to utilise the existing endpoint to elevate the load from them.

Comment (lvrach) [reply config/backend-config/hosted_workspaces.go:84]: cc @fxenik

**Pull request merged**